app/controllers/general.php

```diff
@@ -92,6 +103,18 @@
     Storage::setDevice('files', new Local(APP_STORAGE_UPLOADS.'/app-'.$project->getId()));
     Storage::setDevice('functions', new Local(APP_STORAGE_FUNCTIONS.'/app-'.$project->getId()));
 
+    /* 
+    * Response format
+    */
+    $responseFormatEnvVar = App::getEnv('_APP_SYSTEM_RESPONSE_FORMAT', '');
+    $responseFormatHeader = $request->getHeader('x-appwrite-response-format', '');
+    $responseFormat = empty($responseFormatHeader) ? $responseFormatEnvVar : $responseFormatHeader;
+    if (empty($responseFormat) || ($filter = mapResponseFormatToClass($responseFormat)) == null) {
+        throw new Exception('No filter available for response format : '.$responseFormat, 404);
+    } else {
+        Response::setFilter($filter);   
+    }
+
```

Comment:
You can simplify this logic by doing something like this:

```php
/* 
 * Response format
 */
$responseFormat = $request->getHeader('x-appwrite-response-format', 
    App::getEnv('_APP_SYSTEM_RESPONSE_FORMAT', ''));

switch($responseFormat) {
    case '0.6.0':
    case '0.6.1':
    case '0.6.2':
        Response::setFilter(new V06());
    break;
}
```

We get the header first or fallback to the env var if no header provided. If no filter is requested we just use the default response models with no parsing. This also help us avoid any check on the server startup and avoid the use of any global functions.

Creating a new instance of the filter on every request shouldn't be significant and the cost of complexity might not worth a singleton anyway.

Reply:
Another option here is to use the native `version_compare` function:

```php
switch($responseFormat) {
    case version_compare ($responseFormat , '0.6.2', '=<') :
        Response::setFilter(new V06());
    break;
}

```

Reply:
Awesome. Ill make these changes

Reply:
@eldadfux 

> `$responseFormat = $request->getHeader('x-appwrite-response-format', 
    App::getEnv('_APP_SYSTEM_RESPONSE_FORMAT', ''));`

This is really elegant! Love this

Reply:
and this 

> `switch($responseFormat) {
    case version_compare ($responseFormat , '0.6.2', '=<') :
        Response::setFilter(new V06());
    break;
}`

❤️